### PR TITLE
Fix TERM environment variable setting

### PR DESCRIPTION
**Pull request by jonas-schievink**
Fixes the issues reported in #988 (at least on Arch Linux).

### CODE REVIEW

File: src/main/kotlin/org/rust/cargo/toolchain/Cargo.kt [hunk 1]
```diff
@@ -73,25 +73,24 @@ class Cargo(
...
-            .withEnvironment(CargoConstants.RUSTC_ENV_VAR, pathToRustExecutable)
-            .withEnvironment(environmentVariables)
 
         // Make output colored
         if ((SystemInfo.isLinux || SystemInfo.isMac)
             && command in COLOR_ACCEPTING_COMMANDS
             && additionalArguments.none { it.startsWith("--color") }) {
             cmdLine
-                .withEnvironment("TERM", "linux")
+                .withEnvironment("TERM", "xterm+256color")
```
Comment (alygin) [inline src/main/kotlin/org/rust/cargo/toolchain/Cargo.kt]: Unfortunately, switching to `xterm+256color` breaks coloring on macs. It looks like we need to provide different terminal types for different platforms.

Comment (jonas-schievink) [inline src/main/kotlin/org/rust/cargo/toolchain/Cargo.kt]: *sigh* I knew it wasn't going to be this easy :)

**Comment by jonas-schievink** [comment]
@alygin Can you (or someone else) take a look at the latest commit I just pushed? I don't have a Mac to test this on. We might want to set `TERM` to a value that makes more sense on Macs.

**Comment by alygin** [comment]
It works on my mac

File: src/main/kotlin/org/rust/cargo/toolchain/Cargo.kt (hunk 1 above)
Comment (alygin) [inline src/main/kotlin/org/rust/cargo/toolchain/Cargo.kt]: `linux` uses the same escape codes as `ansi`, while `xterm-256color` differs significantly from them.

BTW, I think we could also try `ansi` as a common ground. It works on mac.

**Comment by matklad** [comment]
Thanks a lot @jonas-schievink and @alygin !

**Pull request merged**